README.md

```diff
@@ -65,8 +65,80 @@ Assets are compiled using Webpack Encore. The compiled assets **must** be commit
 
 ## Installing using Docker
 
-_A new Docker image needs to be created following the move to Wikimedia VPS._
-_You can use the manual installation instructions above in the meantime._
+Development through Docker is suggested if you have a different version of PHP locally
+installed, or if you wish to keep an isolated installation of PHP 8.2 for CopyPatrol.
+
+1. Copy [.env](.env) to [.env.local](.env.local) and fill in the appropriate details.
+   1. Set `REPLICAS_HOST_*` and `TROVE_HOST` to `127.0.0.1`
+   2. Use the credentials in your `replica.my.cnf` file in the home directory of your
+      Toolforge account for `REPLICAS_USERNAME` and `REPLICAS_PASSWORD`.
+   3. Set the rest of the `TROVE_*` variables to that of the installation of the CopyPatrol
+      database (`COPYPATROL_DB_NAME`).
+   4. If you need to test OAuth, obtain tokens by registering a new consumer on Meta at
+      [Special:OAuthConsumerRegistration](https://meta.wikimedia.org/wiki/Special:OAuthConsumerRegistration).
+      Alternatively, you can set `LOGGED_IN_USER` to any value to simulate being that user
+      after clicking on 'Login'.
+   5. If you need to test the "iThenticate report" functionality, set `TCA_DOMAIN` and `TCA_KEY`.
+      Reports older than `AppController::ITHENTICATE_V2_TIMESTAMP` need to connect to the older
+      iThenticate API, using the credentials set by `ITHENTICATE_USERNAME` and `ITHENTICATE_PASSWORD`.
+2. Build the development image once and install Composer dependencies with the following
+   ```bash
+   docker compose build
+   # On Windows, use `%CD%` instead of `$(pwd)`.
+   docker run --rm -ti -v $(pwd):/app wikimedia/copypatrol:development composer install
+   ```
+   Run the second command again every time you change `composer.json`, or when `composer.json`
+   is changed in an upstream commit. This can take a while on Windows.
+3. (*Windows only*) Set the `HOME` environment variable to your user profile directory.
+   ```cmd
+   setx HOME %UserProfile%
+   set HOME=%UserProfile%
+   ```
+   The first command sets `HOME` for future shells. The second command sets `HOME` for the current shell.
+4. Open a new terminal and start the development container with
+   ```bash
+   docker compose up
```

Comment:
Here I get:
```
[+] Running 1/0
 ✔ Container copypatrol-copypatrol-1  Recreated                                                                                                                                                                             0.0s 
Attaching to copypatrol-1
Gracefully stopping... (press Ctrl+C again to force)
Error response from daemon: failed to create task for container: failed to create shim task: OCI runtime create failed: runc create failed: unable to start container process: exec: "/docker-entrypoint.sh": permission denied: unknown
```
I am able to use docker without `sudo`, i.e. I've done this: https://docs.docker.com/engine/install/linux-postinstall/#manage-docker-as-a-non-root-user

Any ideas?

Reply:
Whoops, it looks like I forgot to remove a few lines I was using for testing in the `docker-compose.yml`:
```yml
-   type: bind
    source: "./docker/docker-entrypoint.sh"
    target: "/docker-entrypoint.sh"
    read_only: true
```
Though this *would* work on an OS like Windows where file permissions are mounted as 777 (because it has to go through some translation layer), this would fail on Linux since the permissions on the host machine would be followed (and in this case, `docker-entrypoint.sh` is committed as 666, so it won't have those permissions without it being explicitly given). I was testing the setup out on a Windows PC, so I never noticed that this was still in the Compose config. Will remove this now.